### PR TITLE
fft2

### LINKED ISSUE #12507: fft2

### PR DESCRIPTION
**Pull request by Ali-Ibtehaj-Asif-Niazi**
Close #12507

### CODE REVIEW

**Comment by Ali-Ibtehaj-Asif-Niazi** [comment]
Hi, I have some questions

File: ivy/functional/frontends/numpy/fft/discrete_fourier_transform.py [hunk 1]
```diff
@@ -1,5 +1,7 @@
 import ivy
+import numpy as np
```
Comment (juliagsy) [inline ivy/functional/frontends/numpy/fft/discrete_fourier_transform.py:2]: native numpy should not be a dependency in the ivy frontend

File: ivy/functional/frontends/numpy/fft/discrete_fourier_transform.py [hunk 2]
```diff
@@ -8,3 +10,20 @@ def ifft(a, n=None, axis=-1, norm=None):
...
+
+
+def fft2_layer(
+        x: np.ndarray,
+        s: Optional[Tuple[int, int]] = None,
+        axes: Tuple[int, int] = (-2, -1),
+        norm: Optional[str] = None) -> np.ndarray:
+    if norm is None:
+        norm = "backward"
+    return np.fft.fft2(x, s=s, axes=axes, norm=norm)
```
Comment (juliagsy) [inline ivy/functional/frontends/numpy/fft/discrete_fourier_transform.py:23]: numpy native fit function should not be used here

File: ivy/functional/frontends/numpy/fft/discrete_fourier_transform.py [hunk 3]
```diff
@@ -8,3 +10,20 @@ def ifft(a, n=None, axis=-1, norm=None):
...
+
+@to_ivy_arrays_and_back
+def fft2(a, s=None, axes=(-2, -1), norm=None):
+    a = ivy.array(fft2_layer(a, s=s, axes=axes, norm=norm), dtype=ivy.complex128)
```
Comment (juliagsy) [inline ivy/functional/frontends/numpy/fft/discrete_fourier_transform.py:27]: you may either complete this function as a composition of other functions, or in other cases, propose the `fft2` function to be added to the ivy functional API if you see fit, more information here in the section! 
https://lets-unify.ai/ivy/contributing/open_tasks.html#frontend-apis

File: ivy/functional/frontends/numpy/fft/discrete_fourier_transform.py [hunk 4]
```diff
@@ -8,3 +10,20 @@ def ifft(a, n=None, axis=-1, norm=None):
     if norm is None:
         norm = "backward"
     return ivy.ifft(a, axis, norm=norm, n=n)
+
+
+
+def fft2_layer(
+        x: np.ndarray,
+        s: Optional[Tuple[int, int]] = None,
```
Comment (juliagsy) [inline ivy/functional/frontends/numpy/fft/discrete_fourier_transform.py:18]: typehints are not required in the frontend

File: ivy/functional/frontends/numpy/fft/discrete_fourier_transform.py [hunk 5]
```diff
@@ -1,5 +1,7 @@
 import ivy
+import numpy as np
 from ivy.functional.frontends.numpy.func_wrapper import to_ivy_arrays_and_back
+from typing import Optional, Tuple
```
Comment (juliagsy) [inline ivy/functional/frontends/numpy/fft/discrete_fourier_transform.py:4]: this is then not needed as type hints are not required here

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hello! Apologies if the PR is not ready yet, but just thought that I'd leave some side notes first so that you're in the right direction! The comments are left below, thanks again for the hard work!

**Pull request closed without merging**